resources/lib/ui/utils.py

```diff
@@ -13,6 +14,9 @@ def fetch_sources(sources, dialog, raise_exceptions=False):
     fetched_sources = []
```

Comment:
fetch_sources should accept so maximum length (default is 0 which means infinite)
then, if maximum sources found, just return the list as a dict, there is no reason to break twice for it.